modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetastorageDivergedException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.metastorage.impl;
+
+import org.apache.ignite.internal.lang.IgniteInternalException;
+import org.apache.ignite.lang.ErrorGroups;
+
+/**
+ * Thrown when Metastorage on a node doing Metastorage reentry has been found to be diverged with the leader.
+ */
+public class MetastorageDivergedException extends IgniteInternalException {
```

Comment:
Is this exception expected to be propagated to the outside world?

Reply:
User may get it wrapped in other exceptions which are public, but this one is a technical detail, IMO

Reply:
I'm confused about this. Should it really be an internal exception? This is not a bug or something, but we also don't handle this exception in any way in the code. Should the user deal with it? If yes, then it is no longer internal

Reply:
The only way the user can deal with it now is see it in logs and realize they have to clean the node to use it again

Reply:
How is this supposed to work in the ideal world then?

Reply:
I think that the user will see that the node cannot be started, they will inspect the logs and find the reason: that Metastorage has diverged. There will be a recommendation (in the exception message) to use some kind of a tool (which we don't have now) to resque data, for example, switch to the maintenance mode.

Reply:
I added an error code